src/test/java/com/fittracker/fittracker/service/WeightServiceTest.java

```diff
@@ -0,0 +1,92 @@
+package com.fittracker.fittracker.service;
+
+import com.fittracker.fittracker.entity.Weight;
+import com.fittracker.fittracker.exception.WeightAlreadyExistsException;
+import com.fittracker.fittracker.exception.WeightNotFoundException;
+import com.fittracker.fittracker.repository.WeightRepository;
+import com.fittracker.fittracker.request.WeightRequest;
+import com.fittracker.fittracker.response.WeightResponse;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class WeightServiceTest {
+
+    @Mock
+    private WeightRepository weightRepository;
+
+    @InjectMocks
+    private WeightService weightService;
+
+    private final Weight weight = new Weight(LocalDate.of(2023, 10, 10), 100.1);
+
+    private final WeightRequest weightRequest = new WeightRequest(LocalDate.of(2023,10,10),100.1);
+
+    @Nested
+    class FindByDate {
+        @Test
+        void givenDateFound_shouldReturnWeightResponse() {
+            when(weightRepository.findByDate(any())).thenReturn(Optional.of(weight));
+            var expectedWeightResponse = WeightResponse.fromWeight(weight);
+            var result = weightService.findByDate(LocalDate.of(2023, 10, 10));
```

Comment:
we can extract `LocalDate.of(2023, 10, 10)` to private field, it's used in a few places